light.lua

```diff
@@ -55,7 +55,7 @@ for i=0,14,1 do
 						local meta = minetest.get_meta(pos)
 						if meta:get_string("channel") ~= channel then return end
 						local value = tonumber(msg)
-						if (not value) or value > 14 or value < 0 then return end
+						if not (value and value <= 14 and value >= 0) then return end
```

Comment:
Alternatively, to make it more obvious this is checking for NaN:
```suggestion
						if not value or value > 14 or value < 0 or minetest.is_nan(value) then return end
```

Reply:
Extra condition with additional function call is kinda pointless?
To me, "abort if not within the range" is clearer...
Or maybe a positive check is even better: `if value and value >= 1 and value <= 14 then ... swap() end`

Reply:
Either way is fine, I was just thinking of the possibility of someone refactoring this and "simplifying" it, and having the bug resurface because they didn't know it was also checking for NaN values.

Reply:
a comment would solve that too (bug resurfacing)

Reply:
I should have left a comment in the code... Added one. (+ one for the timer.) Sidenode: NodeTimers don't crash with NaN, but I wouldn't rely in this